src/array/cpu/spmat_op_impl_coo.cc

```diff
@@ -296,14 +297,14 @@ template COOMatrix COOTranspose<kDLCPU, int64_t>(COOMatrix coo);
 
 ///////////////////////////// COOToCSR /////////////////////////////
 
-// complexity: time O(NNZ), space O(1)
+// complexity: time O(NNZ), space O(NNZ)
```

Comment:
The space complexity is `O(num_threads * N)` right? Could you run the recently merged [benchmark code](https://github.com/dmlc/dgl/blob/master/benchmarks/benchmarks/api/bench_format_conversion.py) and compare the memory consumption and speed up with the previous implementation (especially on those large graphs)? You could copy-paste out the benchmark code and run it locally without the asv pipeline.

Reply:
I updated the complexity in d95a08d to:
```
complexity: time O(NNZ), space O(1) if the coo is row sorted,
time O(NNZ/p + N), space O(NNZ + N*p) otherwise, where p is the number of
threads.
```

I ran the benchmark by just reverting the 

For reddit, the previous implementation averaged 6.63s, and used 6.20 GB of memory.
With the new implementation using OMP_NUM_THREADS=1, it averaged 6.5s and used 6.20 GB of memory.
With OMP_NUM_THREADS=12, it averaged 1.83s and took 6.22 GB of memory.

For ogb-products the previous implementation averaged 11.9s and used 7.15GB of memory.
With the new implementation  using OMP_NUM_THREADS=1, it averaged 12.8s and used 7.15 GB of memory.
With OMP_NUM_THREADS=12, it averaged 2.07s and used 7.36 GB of memory.

For friendster, the previous implementation averaged 13.1s and used 58.6 GB of memory.
With the new implementation using OMP_NUM_THREADS=1, it averaged 22s and used 87.8 GB of memory.
With OMP_NUM_THREADS=12, it averaged 7.32s and used 98 GB of memory.
Note, friendster is sorted, and thus the original implementation takes advantage of re-using arrays to reduce memory consumption.

To get the memory usage I ran the commands with `/usr/bin/time -v`, and the above is the `Maximum resident set size`.

Reply:
Cool! Thanks for the benchmarks. It clears my concern.